src/steam/SteamHandler.cs

```diff
@@ -0,0 +1,192 @@
+﻿using System;
+using System.Linq;
+using System.Reflection;
+using Godot;
+
+/// <summary>
+///   Handles communicating between the general codebase and the Steam sdk requiring code.
+///   This ensures that non Steam versions of the game can be built with normal Godot tools.
+///   If Steam library is not loaded most methods in here do nothing to allow easily calling them without checking
+///   first.
+/// </summary>
+public class SteamHandler : Node, ISteamSignalReceiver
+{
+    private static SteamHandler instance;
+
+    private bool wePaused;
+
+    private ISteamClient steamClient;
+
+    public SteamHandler()
+    {
+        instance = this;
+    }
+
+    public static SteamHandler Instance => instance;
+
+    /// <summary>
+    ///   True if Steam has been properly loaded
+    /// </summary>
+    public bool IsLoaded => steamClient is { IsLoaded: true };
+
+    /// <summary>
+    ///   True if Steam loading was attempted and error messages can be retrieved
+    /// </summary>
+    public bool WasLoadAttempted { get; private set; }
+
+    public string DisplayName
+    {
+        get
+        {
+            ThrowIfNotLoaded();
+            return steamClient.DisplayName;
+        }
+    }
+
+    /// <summary>
+    ///   Checks if current running code is an exported Thrive version for Steam
+    /// </summary>
+    /// <returns>Returns true if this is the Steam export</returns>
+    /// <remarks>
+    ///   <para>
+    ///     This is different from detecting if Steam features are loaded, as correct license information must be
+    ///     displayed even if Steam loading fails, for example.
+    ///   </para>
+    /// </remarks>
+    public static bool IsTaggedSteamRelease()
+    {
+        return OS.HasFeature("steam");
+    }
+
+    public override void _Ready()
+    {
+        PauseMode = PauseModeEnum.Process;
+
+        OnSteamInit();
+    }
+
+    public override void _Process(float delta)
+    {
+        steamClient?.Process(delta);
+    }
+
+    /// <summary>
+    ///   Should only be called by the Steam handling library when its loaded
+    /// </summary>
+    /// <param name="client">The Steam handler to register</param>
+    /// <exception cref="ArgumentException">If client is invalid</exception>
+    /// <exception cref="InvalidOperationException">If a client has already registered</exception>
+    public void RegisterSteamClient(ISteamClient client)
```

Comment:
I'd make this internal just to show even clearer that this is not meant for public use.

Reply:
In my original design this actually *wouldn't* be possible to be internal, as this would be called from a separately loaded DLL...
So this kinda isn't Thrive library internal method.